confgenerator/otel/processors.go

```diff
@@ -602,3 +602,14 @@ func GCPProjectID(projectID string) Component {
 		map[string]string{"gcp.project_id": projectID}, false,
 	)
 }
+
+func MetricUnknownCounter() Component {
```

Comment:
nit: we should leave a comment here for breadcrumbs for why we needed this. Maybe point to a golink to our design doc?